### DISCUSSION TITLE
Sorting drop down lists by ID instead of description

### DISCUSSION THREAD

**Discussion by shortsam1** [serenity-is/Serenity#5630] [Help] [answered]
I am new to serenity and .net core. I am using the latest .net core SDK 5.0 and the latest version of serenity. I am trying to sort the month drop down list by the ID field instead of by the description. Any help is greatly appreciated. This is the current drop down list. I would like to sort it by the MonthID field instead.
![image](https://user-images.githubusercontent.com/47765946/114629926-d12cd500-9cfc-11eb-848e-cd968ab2dda1.png)

The tablerow.cs is listed as follows:
`
[DisplayName("Month"), Column("MonthID"), ForeignKey("[online_form].[Ref_Month]", "MonthID"), LeftJoin("jMonth"), TextualField("MonthShortDesc")]
        [LookupEditor(typeof(RefMonthRow))]
        public Int32? MonthId
        {
            get => fields.MonthId[this];
            set => fields.MonthId[this] = value;
        }
`
The monthrow.cs is

`
[DisplayName("Month Id"), Column("MonthID"), Identity, IdProperty, SortOrder(1)]
        public Int32? MonthId
        {
            get => fields.MonthId[this];
            set => fields.MonthId[this] = value;
        }
`

**Comment by minhhungit** [comment, marked as answer]
you can create a custom lookup for yourself, here is sample for .net 5 version

```csharp
[LookupScript("CustomLookup.YourCustomLookup", Expiration = -1, Permission = "*")]
public class YourCustomLookup: Serenity.Web.LookupScript
{
    ISqlConnections _sqlConnnection;
    public YourCustomLookup(ISqlConnections sqlConnnection)
    {
        _sqlConnnection = sqlConnnection;
        this.IdField = "CountryTwoLetterCode";
        this.TextField = "CountryDisplayName";
    }

    protected override IEnumerable<dynamic> GetItems()
    {
        using (var con = _sqlConnnection.NewFor<Entities.CountryRow>())
        {
            var t0 = Entities.CountryRow.Fields;

            var query = new SqlQuery();
            query.From(t0)
                .Select("CountryDisplayName + ' (' + CountryTwoLetterCode + ')'", "CountryDisplayName")
                .Select("CountryTwoLetterCode")
                .Where(t0.CountryTwoLetterCode.IsNotNull())
                .Where(t0.CountryDisplayName.IsNotNull())
                .OrderBy(t0.CountryId);

            return con.Query(query);
        }
    }
}
```

then use it like this:

```csharp
[LookupEditor(typeof(YourCustomLookup)]
public Int32? MonthId { 
  get => fields.MonthId[this]; 
  set => fields.MonthId[this] = value; 
}
```

By this way you can return any result you want

**Reply by axelcasauran** [reply]
is the Sort property/attribute not working?

**Reply by dfaruque** [reply]
Sort property/attribute works in the grid not in the lookup, I think.

**Reply by axelcasauran** [reply]
when you create custom lookupeditor, the "edit" dialog is not working --can you confirm that?

**Reply by shortsam1** [reply]
I'm sorry I cannot remember now as I asked the question nearly a year ago, and I have finished the work and moved into a new role.  Also my question was answered by minhungit.

**Reply by minhhungit** [reply]
edit dialog should work as normal, nothing difference

**Reply by axelcasauran** [reply]
it did not work on me.. hmmm :( 

![2022-02-20 22_48_51-People - HappyPocket](https://user-images.githubusercontent.com/13443821/154908618-bcf588f2-e1e2-41bd-8099-23ec9cf0db59.png)

**Reply by minhhungit** [reply]
open f12 browser and check error log, normally you will need to define dialogType for lookupeditor attribute
